### DISCUSSION TITLE
Adding a config pre/suffix for the template cache key

### DISCUSSION THREAD

**Discussion by ghost** [craftcms/cms#8975] [Ideas]
Currently the cache key is build using only the currentSite id on:
https://github.com/craftcms/cms/blob/8f8916351aaad6132a1c45c8b79530fa1d17fe10/src/services/TemplateCaches.php#L325

We currently are forced to hack this, because our multilingual and multi-domain setup results  in wrongly cached items (every language is accessible on every tld, while having the canonical tld versions referenced in the header).

I just added the baseUrl for our use-case: `$key = $key.Craft::getAlias('@baseUrl');` before the `"template::$key"` stuff.

A clean solution would be a call to a config value at this point.

**Comment by brandonkelly** [comment]
Alternatively you could specify a custom cache key via the [`using key`](https://craftcms.com/docs/3.x/dev/tags.html#using-key) `{% cache %}` tag param, which factors in the base URL.

**Comment by ghost** [comment]
Thanks, that's definitely a way around this issue. But in hot fixing this, re-evaluating all the cache tags was not really an option. Thought creating such a global option would be a nice clean addition.
